src/main/java/book/store/repository/book/BookRepository.java

```diff
@@ -1,8 +1,20 @@
 package book.store.repository.book;
 
 import book.store.model.Book;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    @Query("FROM Book b LEFT JOIN FETCH b.categories с WHERE b.id = :id")
+    Optional<Book> findAllBookInfoById(Long id);
```

Comment:
```suggestion
    Optional<Book> findBookById(Long id);
```

Reply:
Do I have to return DtoWithoutCategories? Because otherwise I get LazyInitializationException, that is why I modified queries.